front/connectors/src/slack/slack.ts

```diff
@@ -0,0 +1,178 @@
+            dustConfig,
+            last_seen_ts
+          );
+        } else {
+          console.log(
+            "Not processing channel because it is not goals.",
+            channel.name,
+            channel.id
+          );
+        }
+      }
+    }
+    next_cursor = c.response_metadata?.next_cursor;
+  } while (next_cursor);
+}
+
+/**
+ * Fetch all messages from a given channel and process only threaded messages (for now).
+ * @param client Slack api web client
+ * @param channel_id channel id (EG: C01B1B2B3C4)
+ * @param last_seen_ts only sync messages created after this timestamp.
+ */
+async function processChannelMessages(
+  client: WebClient,
+  channel: Channel,
+  dustConfig: DustConfig,
+  last_seen_ts?: string
+) {
+  let next_cursor = undefined;
+
+  do {
+    const history: ConversationsHistoryResponse =
+      await client.conversations.history({
+        channel: channel.id!,
+        limit: 100,
+        include_all_metadata: true,
+        cursor: next_cursor,
+        oldest: last_seen_ts,
+      });
+    next_cursor = history.response_metadata?.next_cursor;
+    if (history.error) {
+      throw new Error(history.error);
+    }
+    if (!history.messages) {
+      continue;
+    }
+    for (const message of history.messages) {
+      if (message.thread_ts) {
+        await processMessageThread(
+          client,
+          channel,
+          message.thread_ts,
+          dustConfig
+        );
+      }
+    }
+  } while (next_cursor);
+}
+
+/**
+ * Given a thread, format and upload the data to a dust core data source.
+ * @param client Slack api web client
+ * @param channel_id channel id (eg: C04T6JZ480N)
+ * @param thread_ts the thread id we want to process. Slack mixes id and timestamp, so it's called thread_ts
+ */
+async function processMessageThread(
+  client: WebClient,
+  channel: Channel,
+  thread_ts: string,
+  dustConfig: DustConfig
+) {
+  let next_cursor = undefined;
+  do {
+    const replies = await client.conversations.replies({
+      channel: channel.id!,
+      ts: thread_ts,
+    });
+    console.log(
+      "Should post this to a dust data source using the dust API. replies for thread: ",
+      thread_ts,
+      replies.messages?.map((e) => e.text)
+    );
+    await upsertToDatasource(replies, channel, dustConfig, thread_ts);
+    next_cursor = replies.response_metadata?.next_cursor;
+  } while (next_cursor);
+}
+
+/**
+ * Upsert a Slack thread to a Dust data source.
+ */
+async function upsertToDatasource(
+  replies: ConversationsHistoryResponse,
+  channel: Channel,
+  dustConfig: DustConfig,
+  thread_ts: string
+) {
+  const messages_from_thread = replies.messages?.map((e) => e.text);
+  const str_from_thread = messages_from_thread?.join("\n");
+  if (!str_from_thread) {
+    return;
+  }
+
+  const documentId = `${channel.name}-${thread_ts}`;
+  const dust_url = `http://localhost:3000/api/v1/data_sources/${dustConfig.username}/${dustConfig.datasourceId}/documents/${documentId}`;
```

Comment:
You should use the `URL` env variable here, but not sure we have access to it here?